styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorageTransaction.java

```diff
@@ -92,6 +96,14 @@ public WorkflowId store(Workflow workflow) throws IOException {
     return workflow.id();
   }
 
+  @Override
+  public Optional<Workflow> workflow(WorkflowId workflowId) throws IOException {
+    return DatastoreStorage.getOpt(tx, DatastoreStorage.workflowKey(tx.getDatastore().newKeyFactory(), workflowId))
+        .filter(e -> e.contains(PROPERTY_WORKFLOW_JSON))
```

Comment:
Unneeded now?

Reply:
Probably. Or a missing `PROPERTY_WORKFLOW_JSON` should actually throw an IOException instead of returning an empty optional, since the field is always expected.